### ISSUE TITLE
iview.comm.get_index() fails with 404

### ISSUE THREAD

**Issue by sramsay64** [vadmium/python-iview#24]
`./iview-cli -i` will fail but `./iview-cli -p` works fine. Because `iview.comm.get_keyword(keyword)` works but `iview.comm.get_index()` doesn't. Here's the error that I get:
```
scott@linux ~/bin/python-iview $ ./iview-cli  -i    
Traceback (most recent call last):
  File "./iview-cli", line 303, in <module>
    main()
  File "./iview-cli", line 283, in main
    index()
  File "./iview-cli", line 50, in index
    index = iview.comm.get_index()
  File "/home/scott/bin/python-iview/iview/comm.py", line 114, in get_index
    index_data = maybe_fetch(url, ('application/json',))
  File "/home/scott/bin/python-iview/iview/comm.py", line 48, in maybe_fetch
    return fetch_url(url, type, headers=headers)
  File "/home/scott/bin/python-iview/iview/comm.py", line 31, in fetch_url
    with http_get(session, url, types, headers=all_headers) as http:
  File "/home/scott/bin/python-iview/iview/utils.py", line 224, in http_get
    response = session.open(req)
  File "/usr/lib/python3.4/urllib/request.py", line 461, in open
    response = meth(req, response)
  File "/usr/lib/python3.4/urllib/request.py", line 571, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.4/urllib/request.py", line 499, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.4/urllib/request.py", line 579, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```
this could be fixed by changing code calling `iview.comm.get_index()` to instead call `iview.comm.get_keyword('index')`

**Comment by codebje** [comment]
I think the legacy flash API has been fully removed now.  This hurts batch mode more than anything else, afaict.

**Comment by Gil-W** [comment]
Due to a bit missing on the comment above, this took some trial and error.

I edited def batch_index() and replaced:
    `index = iview.comm.get_index()`
With a new line:
   ` index = iview.comm.get_keyword('index')`

**Comment by gdurbrid** [comment]
I think this needs to be changed in def index() as well.

**Comment by vadmium** [comment]
This is unfortunate. I think changing to use get_keyword('index') could work as a temporary workaround. It is not ideal though, because it requires getting a ~2 MB JSON response (~500 kB compressed) just to list a couple hundred series names.

**Comment by vadmium** [comment]
I pushed the keyword=index workaround (revision 709965a). If you update it should make it work again. Sometimes the server hangs for about 10 s before responding, so I upped the timeout to 30 s.

Ideally I would like to find a better API to get the list of series from. The only other one I know of is `http://iview.abc.net.au/api/index` (used in the “redesign” branch), but it doesn’t use the same decimal series id. numbers, so it is going to need more effort to get everything to work.
